tests/components/tankerkoenig/test_config_flow.py

```diff
@@ -0,0 +1,241 @@
+    ):
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MOCK_USER_DATA
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "user"
+        assert result["errors"][CONF_API_KEY] == "invalid_auth"
+
+
+async def test_user_no_stations(hass: HomeAssistant):
+    """Test starting a flow by user which does not find any station."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+    with patch(
+        "homeassistant.components.tankerkoenig.config_flow.getNearbyStations",
+        return_value={"ok": True, "stations": []},
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MOCK_USER_DATA
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "user"
+        assert result["errors"][CONF_RADIUS] == "no_stations"
+
+
+async def test_import(hass: HomeAssistant):
+    """Test starting a flow by import."""
+    with patch(
+        "homeassistant.components.tankerkoenig.async_setup_entry"
+    ) as mock_setup_entry, patch(
+        "homeassistant.components.tankerkoenig.config_flow.getNearbyStations",
+        return_value=MOCK_NEARVY_STATIONS_OK,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=MOCK_IMPORT_DATA
+        )
+        assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+        assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+        assert result["data"][CONF_NAME] == "Home"
+        assert result["data"][CONF_API_KEY] == "269534f6-xxxx-xxxx-xxxx-yyyyzzzzxxxx"
+        assert result["data"][CONF_FUEL_TYPES] == ["e5"]
+        assert result["data"][CONF_LOCATION] == {"latitude": 51.0, "longitude": 13.0}
+        assert result["data"][CONF_RADIUS] == 2.0
+        assert result["data"][CONF_STATIONS] == [
+            "3bcd61da-xxxx-xxxx-xxxx-19d5523a7ae8",
+            "36b4b812-xxxx-xxxx-xxxx-c51735325858",
+            "3bcd61da-yyyy-yyyy-yyyy-19d5523a7ae8",
+            "36b4b812-yyyy-yyyy-yyyy-c51735325858",
+        ]
+        assert result["options"][CONF_SHOW_ON_MAP]
+
+        await hass.async_block_till_done()
+
+    assert mock_setup_entry.called
+
+
+async def test_options_flow(hass: HomeAssistant):
+    """Test options flow."""
+
+    mock_config = MockConfigEntry(
+        domain=DOMAIN,
+        data=MOCK_USER_DATA,
+        options={CONF_SHOW_ON_MAP: True},
+        unique_id=f"{DOMAIN}_{MOCK_USER_DATA[CONF_LOCATION][CONF_LATITUDE]}_{MOCK_USER_DATA[CONF_LOCATION][CONF_LONGITUDE]}",
+    )
+    mock_config.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.tankerkoenig.async_setup_entry"
+    ) as mock_setup_entry:
+        await mock_config.async_setup(hass)
```

Comment:
Please always use the `hass.config_entries.async_setup` to set up a config entry.